libpkg/pkg_osvf.c

```diff
@@ -0,0 +1,1459 @@
+	{
+		return OSVF_ECOSYSTEM_OPENSUSE;
+	}
+	else if(strncmp(ecosystem, "OSS-Fuzz", 7) == 0)
+	{
+		return OSVF_ECOSYSTEM_OSS_FUZZ;
+	}
+	else if(strncmp(ecosystem, "Packagist", 8) == 0)
+	{
+		return OSVF_ECOSYSTEM_PACKAGIST;
+	}
+	else if(strncmp(ecosystem, "Photon OS", 8) == 0)
+	{
+		return OSVF_ECOSYSTEM_PHOTON_OS;
+	}
+	else if(strncmp(ecosystem, "Pub", 3) == 0)
+	{
+		return OSVF_ECOSYSTEM_PUB;
+	}
+	else if(strncmp(ecosystem, "PyPI", 4) == 0)
+	{
+		return OSVF_ECOSYSTEM_PYPI;
+	}
+	else if(strncmp(ecosystem, "Red Hat", 7) == 0)
+	{
+		return OSVF_ECOSYSTEM_RED_HAT;
+	}
+	else if(strncmp(ecosystem, "Rocky Linux", 10) == 0)
+	{
+		return OSVF_ECOSYSTEM_ROCKY_LINUX;
+	}
+	else if(strncmp(ecosystem, "RubyGems", 8) == 0)
+	{
+		return OSVF_ECOSYSTEM_RUBYGEMS;
+	}
+	else if(strncmp(ecosystem, "SUSE", 4) == 0)
+	{
+		return OSVF_ECOSYSTEM_SUSE;
+	}
+	else if(strncmp(ecosystem, "SwiftURL", 8) == 0)
+	{
+		return OSVF_ECOSYSTEM_SWIFTURL;
+	}
+	else if(strncmp(ecosystem, "Ubuntu", 6) == 0)
+	{
+		return OSVF_ECOSYSTEM_UBUNTU;
+	}
+	else if(strncmp(ecosystem, "Wolfi", 5) == 0)
+	{
+		return OSVF_ECOSYSTEM_WOLFI;
+	}
+
+	return OSVF_ECOSYSTEM_UNKNOWN;
+}
+
+unsigned int
+pkg_osvg_get_reference(const char *reference_type)
+{
+
+	if(strncmp(reference_type, "ADVISORY", 8) == 0)
+	{
+		return OSVF_REFERENCE_ADVISORY;
+	}
+	else if(strncmp(reference_type, "ARTICLE", 5) == 0)
+	{
+		return OSVF_REFERENCE_ARTICLE;
+	}
+	else if(strncmp(reference_type, "DETECTION", 5) == 0)
+	{
+		return OSVF_REFERENCE_DETECTION;
+	}
+	else if(strncmp(reference_type, "DISCUSSION", 5) == 0)
+	{
+		return OSVF_REFERENCE_DISCUSSION;
+	}
+	else if(strncmp(reference_type, "REPORT", 5) == 0)
+	{
+		return OSVF_REFERENCE_REPORT;
+	}
+	else if(strncmp(reference_type, "FIX", 5) == 0)
+	{
+		return OSVF_REFERENCE_FIX;
+	}
+	else if(strncmp(reference_type, "INTRODUCED", 5) == 0)
+	{
+		return OSVF_REFERENCE_INTRODUCED;
+	}
+	else if(strncmp(reference_type, "PACKAGE", 4) == 0)
+	{
+		return OSVF_REFERENCE_PACKAGE;
+	}
+	else if(strncmp(reference_type, "EVIDENCE", 8) == 0)
+	{
+		return OSVF_REFERENCE_EVIDENCE;
+	}
+	else if(strncmp(reference_type, "WEB", 8) == 0)
+	{
+		return OSVF_REFERENCE_WEB;
+	}
+
+	return OSVF_REFERENCE_UNKNOWN;
+}
+
+unsigned int
+pkg_osvg_get_event(const char *reference_type)
+{
+	if(strncmp(reference_type, "SEMVER", 6) == 0)
+	{
+		return OSVF_EVENT_TYPE_SEMVER;
+	}
+	else if(strncmp(reference_type, "ECOSYSTEM", 9) == 0)
+	{
+		return OSVF_EVENT_TYPE_ECOSYSTEM;
+	}
+	else if(strncmp(reference_type, "GIT", 3) == 0)
+	{
+		return OSVF_EVENT_TYPE_GIT;
+	}
+
+	return OSVF_EVENT_TYPE_UNKNOWN;
+}
+
+void
+pkg_osvf_parse_package(struct pkg_audit_package *package, const ucl_object_t *package_obj)
+{
+	/* Parses package structure:
+	   "package": {
+	     "ecosystem": "FreeBSD:ports",
+	     "name": "packagename"
+	    },
+	*/
+
+	package->names->pkgname = xstrdup(ucl_object_tostring(ucl_object_find_key(package_obj, "name")));
```

Comment:
we probably want to check first if ucl_object_find_key actually return something

Reply:
There is function for that one now